plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -5831,6 +5834,26 @@ protected Answer execute(PvlanSetupCommand cmd) {
         return new Answer(cmd, true, "success");
     }
 
+    protected Answer execute(CleanupVMCommand cmd) {
+        VmwareContext context = getServiceContext();
+        VmwareHypervisorHost hyperHost = getHyperHost(context);
+
+        try {
+            VirtualMachineMO vmMo = hyperHost.findVmOnHyperHost(cmd.getVmName());
+            if (vmMo == null) {
+                String msg = String.format("VM [%s] not found on vCenter, cleanup not needed.", cmd.getVmName());
+                s_logger.debug(msg);
+                return new Answer(cmd, true, msg);
+            }
+            vmMo.destroy();
```

Comment:
just curious, why not do it in the process for DestroyCommand ?

Reply:
DestroyCommand centers around the volume. I guess I could have just added conditions around it, but keeping them separate seemed fitting.